### PR TITLE
Update non wagmi webapp packages to major versions 

### PR DESCRIPTION
**Pull request by asmattic**
## Issue

- [Frontend - Update all webapp dependencies to their latest major versions #18](https://github.com/gnars-dao/gnars-frontend/issues/18)

## Notes

- I'll checkout a fresh fork or squash merge in the future to not spam the PR commits. I can fix if needed. [refactor(update-packages): Updating webapp packages, .npmrc](https://github.com/gnars-dao/gnars-frontend/pull/29/commits/f072214966ad3c9fa44aa8bb08ea91b11d18bb8d) is the first commit.
- As requested, I didn't migrate to `next` from `pages` to `app` router here. Next 14 is backwards compatible and designed to be migrated as needed if/when that is something we decide to prioritize. It's not too large a task, but testing for that should be a stand alone effort/PR IMO.

---

## Acceptance Criteria

1. All dependencies are updated to their latest major version.
> `wagmi@v2` contains major breaking changes that are likely dependent on base jump migration. Dependent packages (`viem`, `abitype` etc.) are unchanged.
3. There must be no regression in site functionality.
> No apparent regressions unless the `hardhat` error in `gnar-contracts` persists in Vercel. 

> Without thorough tests in place, the peer/sub dependency graph is basically a turtles all the way down rabbit hole so testing in a Vercel temp build is the best way to test this update.

5. Build and run steps remain simple and consistent with the README.

> *Possible `.graphclient` errors upon runing `pnpm run build:graphclient` manually.* Need to test running manually on current `gnars-dao:main` to determine if this is a regression. The `webapp` build runs successfully with the current `.graphclient`. 

> `pnpm run wagmi:generate` rebuild works without errors.

> No build config changes were made.

7. The site is builds and deploys on Vercel.
> Awaiting deploy authorization to see if there are any build errors (logs). If `hardhat` block number build config error persists in Vercel. That can be addressed.

### CODE REVIEW

**Comment by asmattic** [comment]
All non `wagmi` related packages updated to next major versions.

**Comment by asmattic** [comment]
Not sure exactly where the hardhat error is originating, if it just needs the `.env` keys in Vercel but the `webapp` builds without errors locally. 

image

**Comment by asmattic** [comment]
Screenshot 2024-04-22 at 1 16 15 PM

**Comment by mulfdev** [comment]
yeah no worries, as long as web builds we're good. will review today

**Comment by asmattic** [comment]
All added formatting unrelated to PR removed. Local webapp build confirmed still working.

image

File: .npmrc
Comment (mulfdev) [inline .npmrc:1]: lets remove this

File: .nvmrc
Comment (mulfdev) [inline .nvmrc:1]: per our convo, lets only specify major version

File: package.json
Comment (mulfdev) [inline package.json:1]: only major version of node should be specified

File: packages/webapp/.npmrc
Comment (mulfdev) [inline packages/webapp/.npmrc:1]: remove this file

File: packages/webapp/data/image-data-V2.json
Comment (mulfdev) [inline packages/webapp/data/image-data-V2.json:1]: why is this here?

**Comment by asmattic** [comment]
Removed `.npmrc` files. Kept `node@v18.20.2`. Webapp builds.

image

File: .nvmrc
Comment (asmattic) [reply .nvmrc:1]: Specified node 18 major version only

**Review by asmattic — COMMENTED** [review]
Addressed all requested changes

**Comment by mulfdev** [comment]
the build for webapp does not work presently
CleanShot 2024-04-26 at 19 32 22

**Review by mulfdev — CHANGES_REQUESTED** [review]
CleanShot 2024-04-26 at 19 32 54

**Comment by mulfdev** [comment]
Also let's rebase off the most recent merge

**Comment by asmattic** [comment]
> CleanShot 2024-04-26 at 19 32 54

Hmm, I was getting this error before but it seemed to be only related to upgrading `wagmi` or `viem` versions. I'll rebase and get try to get my `pnpm` environment clean so I can replicate and fix. From what I saw of the recent merge, the chain config in there likely nullifies this issue. I'll push those changes with any additional comments or considerations today.

**Comment by mulfdev** [comment]
Vercel preview is only for the given branch. Didnt see the branch change, let's limit that in the future or close and reopen the pr.

**Comment by asmattic** [comment]
> Vercel preview is only for the given branch. Didnt see the branch change, let's limit that in the future or close and open a new pr.

I'll make sure to have `main` as the comparison fork branch for future PR's. In the interest of not creating a 3rd PR for this issue I think we should wrap up this issue here. 

I just need whatever is causing the `webapp` build error being produced in the Vercel preview build to debug.

**Comment by mulfdev** [comment]
> > Vercel preview is only for the given branch. Didnt see the branch change, let's limit that in the future or close and open a new pr.
> 
> I'll make sure to have `main` as the comparison fork branch for future PR's. In the interest of not creating a 3rd PR for this issue I think we should wrap up this issue here.
> 
> I just need preview build logs to locate and debug the cause of the `webapp` build error produced.

CleanShot 2024-04-29 at 16 40 09

**Comment by mulfdev** [comment]
for commit hash [ae732f9](https://github.com/asmattic/gnars-frontend/commit/ae732f9f0a5717cb37f638604db1570dab09a7f3)

CleanShot 2024-04-29 at 16 45 03


the vercel logs dont reveal anything here that wasnt determinable on local?

**Review by mulfdev — CHANGES_REQUESTED** [review]
326596999-6d16df5e-777d-47c9-bfa7-d4e8318f238a

**Comment by asmattic** [comment]
> ae732f9

Thanks, I'll debug from there and try and replicate locally. I'll request re-review when I have tracked down the root cause.

**Comment by asmattic** [comment]
Dependency `connectkit@^1.5.1` was pulling in `connectkit@1.7.x` (see breaking changes here)[https://docs.family.co/connectkit/migration-guide#section-170-breaking-changes-update-packages] which requires `wagmi@v2`. The `alchemyId` error was an artifact of the `wagmi@v2` larger upgrade due to the `connectkit` relation.  

Replicated the `alchemyId` type error removing `pnpm-lock.yaml` in root and running `pnpm install` and `build` in `webapp` below where you can see the `connectkit` breaking version.

connectkit-1 7 x depends on wagmiv2- 2024-05-01 at 2 00 00 PM

Fixing `connectkit` to `v1.5.1` resolves this issue and runs a successful build. I'll post another local build screenshot, lol. 

I'm pretty confident if another build error occurs, it'll at least be a new one.

**Comment by mulfdev** [comment]
CleanShot 2024-05-01 at 13 29 02
🔥

**Pull request merged**